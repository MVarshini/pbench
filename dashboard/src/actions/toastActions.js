import * as TYPES from "./types";

export const showSessionExpired = () => async (dispatch) => {
  const toast = {
    variant: "danger",
    title: "Session Expired",
    message: "Please login to continue",
  };
  dispatch({
    type: TYPES.SHOW_TOAST,
    payload: toast,
  });
  dispatch(logout());
};

export const showFailureToast = () => async (dispatch) => {
  const toast = {
    variant: "danger",
    title: "Something went wrong",
    message: "Please try again later",
  };
  dispatch({
    type: TYPES.SHOW_TOAST,
    payload: toast,
  });
};

export const constructToast = (variant, title, message = "") => {
  return {
    type: TYPES.SHOW_TOAST,
    payload: { variant, title, message },
  };
};
