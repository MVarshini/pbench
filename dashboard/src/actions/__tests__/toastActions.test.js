import * as ACTIONS from "../toastActions";

// import configureStore to create a mock store where we will dispatch our actions
import configureStore from "redux-mock-store";
// import thunk middle to make our action asynchronous
import thunk from "redux-thunk";

// initialise middlewares
const middlewares = [thunk];
// initialise MockStore which is only the configureStore method which take middlewares as its parameters
const mockStore = configureStore(middlewares);
const mockState = {
  alerts: [
    {
      variant: "success",
      title: "Logged in successfully!",
      message: 112,
      key: expect.anything(),
    },
    {
      variant: "success",
      title: "Saved!",
      message: 114,
      key: expect.anything(),
    },
  ],
};
const store = mockStore(mockState);
beforeAll(() => {
  // making getState as mock function and returning mock value
  store.getState = jest.fn().mockReturnValue(mockState);
});

afterAll(() => {
  jest.clearAllMocks();
  jest.resetAllMocks();
});

test("Show showFailureToast", () => {
  store.dispatch(ACTIONS.showFailureToast());

  expect(store.getActions()).toMatchSnapshot();
});

test("Show Toast", () => {
  store.dispatch(ACTIONS.showToast("success", "Logged in successfully!"));
  expect(store.getActions()).toMatchSnapshot();
});
