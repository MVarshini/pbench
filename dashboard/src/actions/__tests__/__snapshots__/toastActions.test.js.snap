// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Toast 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "key": "l9ipe44gmphh5c6her",
        "message": "Please try again later",
        "title": "Something went wrong",
        "variant": "danger",
      },
    ],
    "type": "SHOW_TOAST",
  },
  Object {
    "payload": Array [
      Object {
        "key": "l9ipe44pc1hk7g81kjh",
        "message": "",
        "title": "Logged in successfully!",
        "variant": "success",
      },
    ],
    "type": "SHOW_TOAST",
  },
]
`;

exports[`Show showFailureToast 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "key": "l9ipe44gmphh5c6her",
        "message": "Please try again later",
        "title": "Something went wrong",
        "variant": "danger",
      },
    ],
    "type": "SHOW_TOAST",
  },
]
`;
